Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/RenamingPolicy.cs

```diff
@@ -1,270 +1,249 @@
+
+        if (string.IsNullOrEmpty(remoteFileDefinition))
         {
-            if (!string.IsNullOrEmpty(remoteFileDefinition) && remoteFileDefinition.Contains("?"))
-            {
-                throw new ArgumentException("Character '?' not allowed in remote filename.", "remoteFileDefinition");
-            }
+            return originalFileNameWithoutPath;
+        }
 
-            if (string.IsNullOrEmpty(originalFileName))
-            {
-                throw new ArgumentException("Original filename must be set.", "originalFileName");
-            }
+        if (!IsFileMask(remoteFileDefinition) &&
+            !IsFileMacro(remoteFileDefinition, MacroHandlers) &&
+            !IsFileMacro(remoteFileDefinition, SourceFileNameMacroHandlers))
+        {
+            // remoteFileDefination does not have macros
+            var remoteFileName = Path.GetFileName(remoteFileDefinition);
 
-            var originalFileNameWithoutPath = Path.GetFileName(originalFileName);
+            if (string.IsNullOrEmpty(remoteFileName))
+                remoteFileDefinition = Path.Combine(remoteFileDefinition, originalFileNameWithoutPath);
 
-            if (string.IsNullOrEmpty(remoteFileDefinition))
-            {
-                return originalFileNameWithoutPath;
-            }
+            return remoteFileDefinition;
+        }
 
-            if (!IsFileMask(remoteFileDefinition) &&
-                !IsFileMacro(remoteFileDefinition, MacroHandlers) &&
-                !IsFileMacro(remoteFileDefinition, SourceFileNameMacroHandlers))
-            {
-                // remoteFileDefination does not have macros
-                var remoteFileName = Path.GetFileName(remoteFileDefinition);
+        var result = this.ExpandMacrosAndMasks(originalFileName, remoteFileDefinition);
 
-                if (string.IsNullOrEmpty(remoteFileName))
-                {
-                    remoteFileDefinition = Path.Combine(remoteFileDefinition, originalFileNameWithoutPath);
-                }
+        if (result.EndsWith("\\"))
+            result = Path.Combine(result, originalFileNameWithoutPath);
 
-                return remoteFileDefinition;
-            }
+        return result;
+    }
 
-            var result = this.ExpandMacrosAndMasks(originalFileName, remoteFileDefinition);
+    public string ExpandDirectoryForMacros(string directory)
+    {
+        if (directory.Contains("%SourceFileName%") || directory.Contains("%SourceFileExtension%"))
+            throw new Exception("'%SourceFileName%' and '%SourceFileExtension%' are not supported macros for source and destination directories.");
 
-            if (result.EndsWith("\\"))
-            {
-                result = Path.Combine(result, originalFileNameWithoutPath);
-            }
+        return ExpandFileMacros(directory);
+    }
+       
+    public string CreateRemoteFileNameForMove(string sourceOperationTo, string sourceFilePath)
```

Comment:
You are not using this method, so it can be removed.

Reply:
Removed.